src/main/jbake/content/blog/2017/12/running-agent.adoc

```diff
@@ -0,0 +1,155 @@
+build from source or grab from a Maven repository the Maven artifact "org.hawkular.agent:hawkular-javaagent-wildfly-dist".
+Look at its `bin/` directory for the domain.conf and standalone.conf files and look in the
+`standalone/configuration` and `domain/configuration` directories for the different agent configuration files. Note also
+the `host-hawkular.xml` that is to be used with the `--host-config` option when running in domain mode.
+
+== How It Works
+
+When the agent starts, it reads its configuration file from disk. The agent then attempts to connect to the Hawkular Server.
+The agent will download some additional configuration from the Hawkular Server that it overlays ontop of its existing
+configuration (this pulls down specific metadata for the types of resources it is to manage).
+
+The agent will also pull down a separate configuration file that is used to configure
+the https://github.com/prometheus/jmx_exporter[Prometheus JMX Exporter] which is started by the agent. This is
+called the "Metrics Exporter". Thus you will get an agent that collects inventory and stores it in Hawkular,
+but you will also have a metrics exporter that will be used to expose metrics for collection by Hawkular's Prometheus Server.
+
+The agent determines what configuration files to pull down via the settings in the local agent configuration file:
+
+```
+subsystem:
+  type-version: WF10
+...
+metrics-exporter:
+  config-file: WF10
+```
+
+As an example, if the agent is to monitor EAP6, you would replace "WF10" with "EAP6". As more supported products are
+added, additional configuration files are loaded on the Hawkular Server which can then be pulled down by the agent
+by simply changing the values of these `type-version` and `config-file` settings.
+
+NOTE: Note that a WildFly host controller must be told to download the metrics exporter
+config file of "WF10-DOMAIN" while the type-version should still be "WF10".
+See the WildFly+Agent distro (Maven artifact org.hawkular.agent:hawkular-javaagent-wildfly-dist) as an example.
+
+When the agent itself is registered in inventory on the Hawkular Server, Hawkular's Prometheus Server will be told to
+scrape the agent's metrics exporter endpoint, thus the agent's metrics are collected automatically.
+
+=== How It Works with WildFly Domain Mode
+
+The agent runs in the host controller. The host controller's agent can get all of the inventory for the host controller
+and its slave servers. There is also a metrics exporter running in the host controller's agent as well (so
+metrics for the host controller can be collected).
+
+An agent runs in each slave server as well, but those agents run in "metrics-only" mode. No inventory is collected
+by these agents, but they do start a metrics exporter so metrics from each slave server can be collected.
+
+To configure the Prometheus JMX Exporter in the slave server's agent, you must turn it on in "slave proxy mode" via these
+settings within the agent configuration file:
```

Comment:
Again, specifics for where to find this file.

Reply:
see above - there is no one place to find this file

Reply:
will add something like: 

"This is the agent configuration file you told the slave server to use in their -javaagent command line argument. See the host.xml entry you added for the slave servers"